view/frontend/web/js/view/payment/method-renderer/omise-offsite-installment-method.js

```diff
@@ -139,7 +139,7 @@ define(
              * @return {integer}
              */
             getTotal: function () {
-                return + window.checkoutConfig.totalsData.grand_total;
+                return + window.checkoutConfig.quoteData.grand_total;
```

Comment:
✋ just wonder what's the difference between `totalsData` and `quoteData`?
Should we also replace all the other places that are using `totalsData` as well?

Reply:
In `totalsData`, _base_grand_total_ and _grand_total_ have different values ideally it should be the same for single currency setup. While on the other hand, `quoteData` has values related to quote such as order prices, customer info. It seems like `quoteData` having correct _grand_total_ value so changing it from `totalsData`.

It will be a good idea to change in other parts as well.

Reply:
So, you're saying `totalsData` has incorrect values? If that's so, what use is it?

Reply:
✋  I just dug into Magento Checkout module (under `vendor/magento/` directory) and found some interesting code.

`module-tax/view/frontend/web/template/checkout/summary/grand-total.html` 👇 
```html
<tr class="grand totals incl">
    <th class="mark" scope="row">
        <strong data-bind="i18n: inclTaxLabel"></strong>
    </th>
    <td data-bind="attr: {'data-th': inclTaxLabel}" class="amount">
        <strong><span class="price" data-bind="text: getValue()"></span></strong>
    </td>
</tr>
```
`grand-total.html` displays its price value using `getValue()` function from `module-tax/view/frontend/web/js/view/checkout/summary/grand-total.js` 👇 

```js
define([
    'Magento_Checkout/js/view/summary/abstract-total',
    'Magento_Checkout/js/model/quote',
    'Magento_Catalog/js/price-utils',
    'Magento_Checkout/js/model/totals'
], function (Component, quote, priceUtils, totals) {
    'use strict';

    return Component.extend({
        defaults: {
            isFullTaxSummaryDisplayed: window.checkoutConfig.isFullTaxSummaryDisplayed || false,
            template: 'Magento_Tax/checkout/summary/grand-total'
        },
        totals: quote.getTotals(),

        ...

        /**
         * @return {*|String}
         */
        getValue: function () {
            var price = 0;

            if (this.totals()) {
                price = totals.getSegment('grand_total').value;
            }

            return this.getFormattedPrice(price);
        },
```
The `totals` variable coming from `Magento_Checkout/js/model/quote` 👇 
```js
define([
    'ko',
    'underscore',
    'domReady!'
], function (ko, _) {
    'use strict';

    /**
     * Get totals data from the extension attributes.
     * @param {*} data
     * @returns {*}
     */
    var proceedTotalsData = function (data) {
            if (_.isObject(data) && _.isObject(data['extension_attributes'])) {
                _.each(data['extension_attributes'], function (element, index) {
                    data[index] = element;
                });
            }

            return data;
        },
        billingAddress = ko.observable(null),
        shippingAddress = ko.observable(null),
        shippingMethod = ko.observable(null),
        paymentMethod = ko.observable(null),
        quoteData = window.checkoutConfig.quoteData,
        basePriceFormat = window.checkoutConfig.basePriceFormat,
        priceFormat = window.checkoutConfig.priceFormat,
        storeCode = window.checkoutConfig.storeCode,
        totalsData = proceedTotalsData(window.checkoutConfig.totalsData),
        totals = ko.observable(totalsData),
        collectedTotals = ko.observable({});

    return {
        totals: totals,
        ...
}
```
Which, referring to `window.checkoutConfig.totalsData`:
```js
totalsData = proceedTotalsData(window.checkoutConfig.totalsData),
totals = ko.observable(totalsData),
```

---

Also, from the code at `vendor/magento/module-checkout/Model/DefaultConfigProvider.php` 👇 

```php
/**
 * Retrieve quote data
 *
 * @return array
 */
private function getQuoteData()
{
    $quoteData = [];
    if ($this->checkoutSession->getQuote()->getId()) {
        $quote = $this->quoteRepository->get($this->checkoutSession->getQuote()->getId());
        $quoteData = $quote->toArray();
        $quoteData['is_virtual'] = $quote->getIsVirtual();

        if (!$quote->getCustomer()->getId()) {
            /** @var $quoteIdMask \Magento\Quote\Model\QuoteIdMask */
            $quoteIdMask = $this->quoteIdMaskFactory->create();
            $quoteData['entity_id'] = $quoteIdMask->load(
                $this->checkoutSession->getQuote()->getId(),
                'quote_id'
            )->getMaskedId();
        }
    }
    return $quoteData;
}
```

and 
```php
private function getTotalsData()
{
    /** @var \Magento\Quote\Api\Data\TotalsInterface $totals */
    $totals = $this->cartTotalRepository->get($this->checkoutSession->getQuote()->getId());
    $items = [];
    /** @var  \Magento\Quote\Model\Cart\Totals\Item $item */
    foreach ($totals->getItems() as $item) {
        $items[] = $item->__toArray();
    }
    $totalSegmentsData = [];
    /** @var \Magento\Quote\Model\Cart\TotalSegment $totalSegment */
    foreach ($totals->getTotalSegments() as $totalSegment) {
        $totalSegmentArray = $totalSegment->toArray();
        if (is_object($totalSegment->getExtensionAttributes())) {
            $totalSegmentArray['extension_attributes'] = $totalSegment->getExtensionAttributes()->__toArray();
        }
        $totalSegmentsData[] = $totalSegmentArray;
    }
    $totals->setItems($items);
    $totals->setTotalSegments($totalSegmentsData);
    $totalsArray = $totals->toArray();
    if (is_object($totals->getExtensionAttributes())) {
        $totalsArray['extension_attributes'] = $totals->getExtensionAttributes()->__toArray();
    }
    return $totalsArray;
}
```
If we dig down to `cartTotalRepository`, it seems both are  initiating the Quote repository the same `vendor/magento/module-quote/Model/GuestCart/GuestCartRepository.php` 👇 
```php
public function get($cartId)
{
    /** @var $quoteIdMask QuoteIdMask */
    $quoteIdMask = $this->quoteIdMaskFactory->create()->load($cartId, 'masked_id');
    return $this->quoteRepository->get($quoteIdMask->getQuoteId());
}
```

---
I'm not sure where it will lead us to... but if the origin of these 2 variable (window.checkoutConfig.quoteData) and (window.checkoutConfig.totalsData) are coming from Quote object, then probably it should return the same result? 🤔 

I was also searching and found no place in Magento project that access to `window.checkoutConfig` directly too. Maybe using data from `window.checkoutConfig` directly might cause this issue? (not sure...)

What if we check what is inside `quote.totals()` or, simply using the price from `Magento_Checkout/js/model/quote` instead of accessing directly to the global `window.checkoutConfig` variable?

Reply:
Sorry for the long comment 🙇

Reply:
Btw, also at `vendor/magento/module-tax/view/frontend/web/js/view/checkout/summary/grand-total.js`
Not sure how it works but looking on these 2 functions and it really seems that `totalsData.grand_total` does not include tax.
```js
/**
 * @return {*|String}
 */
getValue: function () {
    var price = 0;

    if (this.totals()) {
        price = totals.getSegment('grand_total').value;
    }

    return this.getFormattedPrice(price);
},

/**
 * @return {*}
 */
getGrandTotalExclTax: function () {
    var total = this.totals();

    if (!total) {
        return 0;
    }

    return this.getFormattedPrice(total['grand_total']);
},
```

Reply:
@jonrandy I would say not expected values more than incorrect values as still don't know the reason why they have excluded tax amount in grand price.